private.php

```diff
@@ -1688,6 +1689,11 @@
 		$mybb->input['check'] = $mybb->get_input('check', MyBB::INPUT_ARRAY);
 		if(!empty($mybb->input['check']))
 		{
+			if(!$mybb->input['fid'])
+			{
+				$mybb->input['fid'] = 1;
+			}
```

Comment:
I've noticed that this is actually the only check for moving a PM. While this is fine for this ticket as it redirects PMs moving to Unread to end up in Inbox, you can move a PM to any other number for a folder even if it does not exist. (negative numbers result in an SQL error)

Shouldn't be that big a change so are you happy to fix in this PR or would you rather split out into a new issue?

Reply:
Why do you expect an out-of-range number from a $_POST call?
Even this check is a fallback and never gonna get hit as we have removed `Unread` option from select control.

Reply:
The simplest case is having a custom folder already, loading the PMs page then later on delete the folder and after that move a PM to the folder from the original page which still has the form loaded from when the folder existed. I’m not sure there’s an argument for allowing invalid input to be accepted, at the very least negative numbers should be rejected.